app/Model/Repository.php

```diff
@@ -59,14 +66,42 @@ public static function getViewers()
         return $viewers;
     }
 
+    public static function setStatus(Build $build, $complete = true)
+    {
+        $buildProperties = new BuildProperties($build);
+        $buildProperties->Fill();
+        if (!array_key_exists('status context', $buildProperties->Properties)) {
+            return;
+        }
+        $context = $buildProperties->Properties['status context'];
+
+        $repositoryService = self::getRepositoryService($build);
+        if ($complete) {
+            $repositoryService->setStatusOnComplete($build, $context);
+        } else {
+            $repositoryService->setStatusOnStart($build, $context);
+        }
+    }
+
+    protected static function getRepositoryService(Build $build)
+    {
+        $project = new Project();
+        $project->Id = $build->ProjectId;
+        $project->Fill();
+
+        $repositoryInterface = self::getRepositoryInterface($project);
+        $client = new HttpClient();
+        return new RepositoryService($repositoryInterface, $client);
+    }
+
     /**
      * @param Project $project
-     * @return GitHub|null
+     * @return RepositoryInterface
      * @throws Exception
      */
-    public static function factory(Project $project)
+    public static function getRepositoryInterface(Project $project)
```

Comment:
No need to change it here but perhaps in the future we can drop the `Interface` part and just use `getRepository`?